### PR TITLE
Use onClick to redirect entire row to hint page

### PR DESCRIPTION
**Pull request by brian-xu**
Clicking the button when it says "CLAIM" redirects to the hint page, clicking when it says "UNCLAIM" keeps the user on the hint list.

### CODE REVIEW

**Comment by brian-xu** [comment]
For some reason this sort of breaks the next/previous buttons on the table, looking for a fix

**Comment by brian-xu** [comment]
should be fixed by https://github.com/TanStack/table/discussions/2731

**Comment by qiaochloe** [comment]
Looks great! Made some changes so that users stay on the hint page when they click CLAIM. It feels more smooth to me.
Also did some minor refactoring, feel free to squash and merge this PR if it looks good after you check over it @brian-xu

**Comment by alex-j-wang** [comment]
> Looks great! Made some changes so that users stay on the hint page when they click CLAIM. It feels more smooth to me. Also did some minor refactoring, feel free to squash and merge this PR if it looks good after you check over it @brian-xu

Might be worth discussing during meeting, I think last year clicking CLAIM auto-redirected to hint-specific page. Saves a click. Could maybe even have UNCLAIM on a hint-specific page redirect to hint table.

**Comment by qiaochloe** [comment]
Sounds good to me. (I will strike for the CLAIM button.)

**Comment by qiaochloe** [comment]
Spoke with Thomas, am now in favor of a button that opens a new tab in the background and switches you to that view.

**Review by alex-j-wang — COMMENTED** [review]
As of the merge, response box stopped showing, should be fixed by most recent commit but linter is complaining, might be a type issue

File: src/app/(admin)/admin/hints/components/hint-page/ResponseBox.tsx [hunk 1]
```diff
@@ -25,7 +25,7 @@ export function ResponseBox({ hint }: { hint: typeof hints.$inferSelect }) {
     );
   }
 
-  if (hint.claimer == currHinter)
+  if (hint.claimer?.id == currHinter)
```
Comment (alex-j-wang) [inline src/app/(admin)/admin/hints/components/hint-page/ResponseBox.tsx:28]: Specifically here

**Comment by alex-j-wang** [comment]
Oh, fixed by #116. Applying the changes here so linter stops complaining. Might want to make response box match button optimism but otherwise I think this branch is pretty much ready to merge.

File: src/app/(admin)/admin/hints/actions.tsx [hunk 2]
```diff
@@ -45,7 +46,10 @@ export async function claimHint(hintId: number) {
 
   if (hint.claimer && hint.claimer !== session.user.id) {
     revalidatePath("/admin/");
-    throw new Error("Hint claimed by " + hint.claimer + ".");
+    const user = await db.query.teams.findFirst({
```
Comment (alex-j-wang) [inline src/app/(admin)/admin/hints/actions.tsx:49]: Might also want to avoid this query if there's a faster way to find claimer display name

**Review by alex-j-wang — COMMENTED** [review]
Think redirecting (instead of new window) is more natural for row click. Also moved CLAIM click window open to server action so it happens after server confirmation.

**Pull request merged**